Sources/AIProxy/Gemini/GeminiGenerateContentRequestBody.swift

```diff
@@ -0,0 +1,236 @@
+//
+//  GeminiGenerateContentRequestBody.swift
+//
+//
+//  Created by Todd Hamilton on 10/14/24.
+//
+
+import Foundation
+
+/// Chat completion request body. See the Gemini reference for available fields.
+/// Contributions are welcome if you need something beyond the simple fields I've added so far.
+/// Docstrings are taken from this reference:
+/// https://ai.google.dev/api/generate-content#v1beta.models.generateContent
+public struct GeminiGenerateContentRequestBody: Encodable {
+
+    // MARK: Required
+
+    /// The model to use for generating the completion. Format: models/{model}
+    public let model: String
+
+    /// The content of the current conversation with the model.
+    public let contents: [Content]
+
+    // MARK: Optional
+
+    /// The name of the content cached to use as context to serve the prediction.
+    public let cachedContent: String?
+
+    /// Configuration options for model generation and outputs.
+    public let generationConfig: GenerationConfig?
+
+    /// A list of unique SafetySetting instances for blocking unsafe content.
+    public let safetySettings: [SafetySetting]?
+
+    /// Developer set system instruction(s). Currently, text only.
+    public let systemInstruction: SystemInstruction?
+
+    /// A list of Tools the Model may use to generate the next response.
+    public let tools: [Tool]?
+
+    /// Tool configuration for any Tool specified in the request.
+    public let toolConfig: ToolConfig?
+
+    // MARK: Initializer
+
+    public init(
+        model: String,
+        contents: [Content],
+        cachedContent: String? = nil,
+        generationConfig: GenerationConfig? = nil,
+        safetySettings:[SafetySetting]? = nil,
+        systemInstruction: SystemInstruction? = nil,
+        tools: [Tool]? = nil,
+        toolConfig: ToolConfig? = nil
+    ) {
+        self.model = model
+        self.contents = contents
+        self.cachedContent = cachedContent
+        self.generationConfig = generationConfig
+        self.safetySettings = safetySettings
+        self.systemInstruction = systemInstruction
+        self.tools = tools
+        self.toolConfig = toolConfig
+    }
+}
+
+/// Struct representing the content of the conversation with the model.
+/// A Content includes a role field (e.g., "user" or "model") and ordered parts that constitute a message.
+public struct Content: Encodable {
```

Comment:
Please place this inside an extension:
```
// MARK: - RequestBody.Content
public extension GeminiGenerateContentRequestBody {
    struct Content {
       // What you have now
    }
}
```

Reason: we don't want `Content` to mean Gemini content throughout the lib. You can imagine the confusion if we allowed all providers to have a public type called `Content`. By nesting it under the namespace, your property types in the `GeminiGenerateContentRequestBody` definition remain the same, and we eliminate a source of confusion / collision. Repeat for other types, so `Part` for example will be:

```
// MARK: - RequestBody.Content.Part
public extension GeminiGenerateContentRequestBody.Content {
   struct Part {...}
```